pkg/virt-api/webhooks/validating-webhook/admitters/pod-eviction-admitter_test.go

```diff
@@ -0,0 +1,353 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2018 Red Hat, Inc.
+ *
+ */
+
+package admitters
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/golang/mock/gomock"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"k8s.io/api/admission/v1beta1"
+	k8sv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes/fake"
+	"k8s.io/client-go/testing"
+
+	virtv1 "kubevirt.io/client-go/api/v1"
+	"kubevirt.io/client-go/kubecli"
+	"kubevirt.io/kubevirt/pkg/testutils"
+	virtconfig "kubevirt.io/kubevirt/pkg/virt-config"
+)
+
+// we can't just match vmi status conditions because there is a timestamp set on them.
+type vmiStatusConditionsMatcher struct {
+	conditions []virtv1.VirtualMachineInstanceCondition
+	msg        string
+}
+
+type patch struct {
+	Op    string                                   `json:"op"`
+	Path  string                                   `json:"path"`
+	Value []virtv1.VirtualMachineInstanceCondition `json:"value"`
+}
+
+func newVmiStatusConditionsMatcher(conditions []virtv1.VirtualMachineInstanceCondition) gomock.Matcher {
+	return &vmiStatusConditionsMatcher{conditions: conditions}
+}
+
+func (m *vmiStatusConditionsMatcher) Matches(obj interface{}) bool {
+	pb, ok := obj.([]byte)
+	if !ok {
+		m.msg = "Expected object of type []byte"
+		return false
+	}
+
+	var patches []patch
+	err := json.Unmarshal(pb, &patches)
+	if err != nil {
+		m.msg = err.Error()
+		return false
+	}
+
+	matchConditions := patches[0].Value
+	if len(matchConditions) != len(m.conditions) {
+		m.msg = fmt.Sprintf("Conditions differ on lenght: got %d expected %d", len(matchConditions), len(m.conditions))
+		return false
+	}
+
+	for i, condition := range m.conditions {
+		mc := matchConditions[i]
+		if mc.Status != condition.Status || mc.Reason != condition.Reason || mc.Type != condition.Type {
+			diff := fmt.Sprintf("Status: got %s expected %s, Reason: got %s expected %s, Type: got %s expected %s",
+				mc.Status, condition.Status, mc.Reason, condition.Reason, mc.Type, condition.Type)
+			m.msg = fmt.Sprintf("Diff at index %d: %s", i, diff)
+			return false
+		}
+	}
+
+	return true
+}
+
+func (m *vmiStatusConditionsMatcher) String() string {
+	return m.msg
+}
+
+var _ = Describe("Pod eviction admitter", func() {
+
+	newClusterConfigWithFeatureGate := func(featureGate string) *virtconfig.ClusterConfig {
+		clusterConfig, cmInformer, _, _ := testutils.NewFakeClusterConfig(&k8sv1.ConfigMap{})
```

Comment:
lets use the `NewFakeClusterConfigUsingKV` instead so that this doesn't need to be updated in the near future